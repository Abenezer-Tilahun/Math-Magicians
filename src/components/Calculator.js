import React from 'react';
import calculate from '../logic/calculate';
import Button from './Button';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      calcObj: {
        total: null,
        next: null,
        operation: null,
      },
    };
    this.onClick = this.onClick.bind(this);
  }

  onClick(buttonName) {
    const { calcObj } = this.state;
    const newCalcObj = { ...calcObj, ...calculate(calcObj, buttonName) };
    this.setState({
      calcObj: newCalcObj,
    });
  }

  render() {
    const { calcObj } = this.state;
    const result = calcObj.next || calcObj.total;
    return (

      <div className="calculator-wrapper">
        <div className="result-section">
          {Number(result)}
        </div>
        <Button value="AC" onClick={this.onClick} />
        <Button value="+/-" onClick={this.onClick} />
        <Button value="%" onClick={this.onClick} />
        <Button value="รท" onClick={this.onClick} />
        <Button value="7" onClick={this.onClick} />
        <Button value="8" onClick={this.onClick} />
        <Button value="9" onClick={this.onClick} />
        <Button value="x" onClick={this.onClick} />
        <Button value="4" onClick={this.onClick} />
        <Button value="5" onClick={this.onClick} />
        <Button value="6" onClick={this.onClick} />
        <Button value="-" onClick={this.onClick} />
        <Button value="1" onClick={this.onClick} />
        <Button value="2" onClick={this.onClick} />
        <Button value="3" onClick={this.onClick} />
        <Button value="+" onClick={this.onClick} />
        <Button value="0" onClick={this.onClick} />
        <Button value="." onClick={this.onClick} />
        <Button value="=" onClick={this.onClick} />
      </div>
    );
  }
}

export default Calculator;
