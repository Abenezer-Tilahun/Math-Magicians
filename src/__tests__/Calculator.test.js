import React from 'react';
import Renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('Renders correctly', () => {
  const tree = Renderer
    .create(<Calculator />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
