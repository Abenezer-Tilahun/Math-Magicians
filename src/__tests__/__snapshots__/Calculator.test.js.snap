// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="calculator-tab-wrapper"
>
  <div
    className="calculator-info"
  >
    <h2>
      Let's do some math!
    </h2>
    <p>
      Use this calculator to do basic Math operation.
    </p>
  </div>
  <div
    className="calculator-wrapper"
  >
    <div
      className="result-section"
    >
      0
      <div
        className="math"
      >
        hi
      </div>
    </div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn btn-operation"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn btn-operation"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn btn-operation"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn btn-operation"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="btn btn-0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn btn-operation"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
